packages/shorebird_cli/lib/src/commands/release/release_command.dart

```diff
@@ -1,66 +1,448 @@
+Each build must have a unique identifier to differentiate it from previous builds.
+It is used to determine whether one build is more recent than another, with higher numbers indicating more recent build.
+On Android it is used as "versionCode".
+On Xcode builds it is used as "CFBundleVersion".''',
+        defaultsTo: '1.0',
+      )
+      ..addFlag(
+        'codesign',
+        help: 'Codesign the application bundle.',
+        defaultsTo: true,
+      )
+      ..addFlag(
+        'dry-run',
+        abbr: 'n',
+        negatable: false,
+        help: 'Validate but do not upload the release.',
+      )
+      ..addOption(
+        exportOptionsPlistArgName,
+        help:
+            '''Export an IPA with these options. See "xcodebuild -h" for available exportOptionsPlist keys.''',
+      )
+      ..addOption(
+        'flutter-version',
+        help: 'The Flutter version to use when building the app (e.g: 3.16.3).',
+      )
+      ..addOption(
+        'android-artifact',
+        help:
+            '''The type of artifact to generate. Only relevant for Android releases.''',
+        allowed: ['aab', 'apk'],
+        defaultsTo: 'aab',
+        allowedHelp: {
+          'aab': 'Android App Bundle',
+          'apk': 'Android Package Kit',
+        },
+      )
+      ..addMultiOption(
+        'platform',
+        abbr: 'p',
+        help: 'The platform(s) to to build this release for.',
+        allowed: ReleaseType.values.map((e) => e.cliName).toList(),
+        // TODO(bryanoltman): uncomment this once https://github.com/dart-lang/args/pull/273 lands
+        // mandatory: true.
+      )
+      ..addOption(
+        'release-version',
+        help: '''
+The version of the associated release (e.g. "1.0.0"). This should be the version
+of the iOS app that is using this module.''',
+      )
+      ..addMultiOption(
+        'target-platform',
+        help: 'The target platform(s) for which the app is compiled.',
+        defaultsTo: Arch.values.map((arch) => arch.targetPlatformCliArg),
+        allowed: Arch.values.map((arch) => arch.targetPlatformCliArg),
+      );
   }
 
+  late final ResolveReleaser _resolveReleaser;
+
   @override
-  String get description => 'Manage your Shorebird app releases.';
+  bool get hidden => true;
+
+  @override
+  String get description =>
+      'Creates a shorebird release for the provided target platforms';
 
   @override
   String get name => 'release';
 
-  static void printConflictingFlutterRevisionError({
-    required String existingFlutterRevision,
-    required String currentFlutterRevision,
-    required String releaseVersion,
-  }) {
-    logger.err(
-      '''
-${styleBold.wrap(lightRed.wrap('A release with version $releaseVersion already exists but was built using a different Flutter revision.'))}
+  @override
+  Future<int> run() async {
+    final releaserFutures =
+        results.releaseTypes.map(_resolveReleaser).map(createRelease);
+
+    for (final future in releaserFutures) {
```

Comment:
As I look through I wonder if we should
 - log which release we are doing now
 - that such release worked
 - etc

Reply:
The releasers handle the logging, which I think addresses these concerns